src/types.rs

```diff
@@ -271,6 +271,52 @@ pub struct NewOrder<'a> {
     pub identifiers: &'a [Identifier],
 }
 
+/// The reason for a certificate revocation
+/// Defined in <https://datatracker.ietf.org/doc/html/rfc5280#section-5.3.1>
+#[allow(missing_docs)]
+pub enum RevocationReason {
+    Unspecified,
+    KeyCompromise,
+    CaCompromise,
+    AffiliationChanged,
+    Superseded,
+    CessationOfOperation,
+    CertificateHold,
+    RemoveFromCrl,
+    PrivilegeWithdrawn,
+    AaCompromise,
+}
+
+/// Payload for a certificate revocation request
+/// Defined in <https://datatracker.ietf.org/doc/html/rfc8555#section-7.6>
+#[derive(Debug, Serialize)]
+pub struct CertificateRevocation {
+    /// URL safe base64 encoded certificate (not a chain)
+    /// The certificate should not contain a header or footer or line breaks
+    pub certificate: String,
+    /// Reason for revocation
+    pub reason: Option<i32>,
+}
+
+impl CertificateRevocation {
+    /// Create a new revocation request from a certificate.
+    /// The certificate cannot be a chain.
+    pub fn new(certificate: &str, reason: Option<RevocationReason>) -> CertificateRevocation {
```

Comment:
If we're going to do a bunch of processing here on the `certificate` `&str`, it should not be a `pub` field. Instead of having this constructor on `CertificateRevocation` (and making the fields private), I would suggest that we add a dependency on rustls-pki-types, take a `CertificateDer` (which can decode from PEM using the rustls-pemfile crate -- which can be handled by the caller) and then base64url encode it during serialization with another custom `Serialize` impl. What do you think?